Assets/Scripts/Models/Settings.cs

```diff
@@ -59,7 +66,13 @@ public static string GetSettingWithOverwrite(string key, string defaultValue)
         }
 
         settingsDict.Add(key, defaultValue);
-        SaveSettings();
+
+        // we have justed altered a setting so we have to set the flag saying their are unsaved settings
+        if (Settings.unsavedSettings == false) {
```

Comment:
Since you copy and paste this block of code on line 71 again on line 109, consider putting it in a seperate method and then calling it in those places.